File: ShockRecorder/app/src/main/java/com/example/shockrecorder/LoginActivity.java
package com.example.shockrecorder;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.Toast;

public class LoginActivity extends AppCompatActivity {
    private EditText et_email;
    private EditText etpassword;
    private ProgressBar progressBar;
    private Button loginbtn;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        loginbtn=findViewById(R.id.loginbtn);
        et_email = findViewById(R.id.etEmail);
        etpassword = findViewById(R.id.etPassword);
        progressBar = findViewById(R.id.progressBar);
        loginbtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                validate(v);
            }
        });
    }
    private void login(String email, String senha) {
        FirebaseHelper.getAuth().signInWithEmailAndPassword(email, senha)
                .addOnCompleteListener(task -> {
                    if (task.isSuccessful()) {
                        startActivity(new Intent(this, MainActivity.class));
                        finish();
                    } else {
                        String erro = FirebaseHelper.validErros(task.getException().getMessage());
                        Toast.makeText(this, erro, Toast.LENGTH_SHORT).show();
                    }
                    progressBar.setVisibility(View.GONE);
                });
    }
    public void validate(View view) {
        String email = et_email.getText().toString();
        String pass = etpassword.getText().toString();

        if (!email.isEmpty()) {
            if (!pass.isEmpty()) {
                progressBar.setVisibility(View.VISIBLE);
                login(email, pass);
            } else {
                et_email.requestFocus();
                et_email.setError("Enter your email");
            }
        } else {
            etpassword.requestFocus();
            et_email.setError("Enter your email");
        }
    }
}

File: Messenger/app/src/main/java/com/example/messenger/register/LoginActivity.kt
package com.example.messenger.register

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.core.view.isInvisible
import androidx.core.view.isVisible
import com.example.messenger.register.MainActivity
import com.example.messenger.R
import com.example.messenger.message.LatestMessageActivity
import com.google.firebase.auth.FirebaseAuth
import kotlinx.android.synthetic.main.activity_login.*
import kotlinx.android.synthetic.main.activity_main.txtemail
import kotlinx.android.synthetic.main.activity_main.txtpassword

class LoginActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        signinbtn.setOnClickListener {
            PerformLogin()
        }
        SingUptxt.setOnClickListener{
            val intent= Intent(this, MainActivity::class.java)
            intent.flags=Intent.FLAG_ACTIVITY_CLEAR_TASK.or(Intent.FLAG_ACTIVITY_NEW_TASK)
            startActivity(intent)
        }
    }
    private fun PerformLogin(){
       wrongemailPass.isInvisible=true
        val email=txtemail.text.toString().trim(){it<=' '}
        val password=txtpassword.text.toString().trim(){it<=' '}
        if(email.isEmpty()||password.isEmpty())
        {
            Toast.makeText(this,"Please Fill the Box", Toast.LENGTH_LONG).show()
            return
        }
        FirebaseAuth.getInstance().signInWithEmailAndPassword(email,password)
            .addOnCompleteListener{
                if(it.isSuccessful)
                {

                    val intent=Intent(this, LatestMessageActivity::class.java)
                    intent.flags=Intent.FLAG_ACTIVITY_CLEAR_TASK.or(Intent.FLAG_ACTIVITY_NEW_TASK)
                    startActivity(intent)
            }
            }
            .addOnFailureListener{
                wrongemailPass.isVisible=true
                Toast.makeText(this,"Incorrect Email or Password!",Toast.LENGTH_SHORT).show()
            }

    }
}